actions/appSettings.tsx

```diff
@@ -0,0 +1,55 @@
+'use server';
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+export type AppSettings = {
+  confirmToolCalls: boolean;
+  browser: BrowserAppSettings;
+};
+
+export type BrowserAppSettings = {
+  headless: boolean;
+  useDefaultSession: boolean;
+};
+
+const defaultAppSettings: AppSettings = {
+  confirmToolCalls: false,
+  browser: {
+    headless: false,
+    useDefaultSession: false,
+  },
+};
+
+export async function getAppSettings(): Promise<AppSettings> {
+  const location = await AppSettingsLocation();
+  if (fs.existsSync(location)) {
+    const AppSettings = fs.readFileSync(location, 'utf-8');
+    return JSON.parse(AppSettings) as AppSettings;
+  }
+  return defaultAppSettings;
+}
+
+export async function setAppSettings(AppSettings: AppSettings) {
+  const location = await AppSettingsLocation();
```

Comment:
It can just be `GPTSCRIPT_SETTINGS_FILE`

Reply:
That's a really really good point. I should've thought of that, great call out. Simplifies the code a lot.